### ISSUE TITLE
Build Warp for Windows

### ISSUE THREAD

**Issue by gagata** [warpdotdev/Warp#204]
Similarly to the Linux support - ability to run Warp on Windows machines. Adding here to track interest.

Please subscribe if you want to be notified as we make progress.

### How important is this feature to you?

Please add a 👍 to the original post at the top to signal that you want this feature, and subscribe if you'd like to be notified. We'll be posting updates on this thread (if/when this is being worked on). 

Please avoid comments with spam-like "+1" or 👍 .

[Feb 2024 Update](https://github.com/warpdotdev/Warp/issues/204#issuecomment-1960342051):
We'll be tracking support for Warp for Linux on WSL (Windows Subsystem for Linux) separately here https://github.com/warpdotdev/Warp/issues/4240
To be clear, we are still working on making a native Warp for Windows (`exe`), which we will post about on this thread later this year so stay tuned and/or [subscribe to get notified](https://www.warp.dev/windows-terminal)!

**Comment by gagata** [comment]
#120 is the Linux version of this ticket

**Comment by vip3rousmango** [comment]
+1 for Windows 10 version!

**Comment by farzadso** [comment]
+1 for Windows (with proper WSL2 support)

**Comment by NoNameProvided** [comment]
I would like to ask everyone who is excited about the Windows version of Warp to refrain from posting +1 as a comment. I assume a bunch of people is subscribed to this issue besides me and we all get notifications for such comments.

If you are want to show support you can by liking the question as ~150 have done so before.

**Comment by elviskahoro** [comment]
@NoNameProvided I'll just lock this issue. This should help with notifications.

**Comment by dannyneira** [comment]
Hey folks, we'll be tracking support for Warp for Linux on WSL (Windows Subsystem for Linux) here https://github.com/warpdotdev/Warp/issues/424

To be clear, we are still working on making a native Warp for Windows `exe`, which we will post about on this thread later this year so stay tuned!

**Comment by Dhyfer1** [comment]
Hi. I want to join the Warp for Windows waitlist but I can't find where to do it.

I clicked on this link https://www.warp.dev/windows-terminal which leads me to https://www.warp.dev/?windows_waitlist=true but here I don't see any field where I can leave my email to sign up for the waiting list, I only see a field where they email the download links for Linux and Mac.

So how can I get on the waitlist?

**Comment by pfeerick** [comment]
So you don't get this?

![image](https://github.com/warpdotdev/Warp/assets/5500713/2f08a19f-47e4-490d-832c-aeedc7f9f2f7)

**Comment by k3davis** [comment]
Not sure if my issue is the same, but mine says "Send download link" not "join waitlist." Maybe you are probing for OS, given I'm visiting that link from my Android phone and not a Windows machine?

**Comment by alexlamgit** [comment]
I do get it now. Seems to be fixed. 👍

Sent from my iPhone

On Mar 5, 2024, at 9:57 PM, Peter Feerick ***@***.***> wrote:

﻿

So you don't get this?

image.png (view on web)<https://github.com/warpdotdev/Warp/assets/5500713/2f08a19f-47e4-490d-832c-aeedc7f9f2f7>

**Comment by Dhyfer1** [comment]
@pfeerick No, I only get this image from my Android phone, the screenshot was taken with desktop mode activated

![temp_Screenshot_20240305-223821](https://github.com/warpdotdev/Warp/assets/50301201/8a612acc-818e-4932-aa99-d6f8b1b8d6b7)

@k3davis I see you also have the same problem as mine. Have you been able to solve it?

@alexlamgit How did you do it?

**Comment by kchawla-pi** [comment]
You probably need to access that page from a Windows machine. Or maybe
changing the browser agent to a Windows one.
Once the website sees you're trying to download Warp from a Windows system,
it should give you the wait-list option.

**Comment by tusharsnx** [comment]
Website should have a platform selector anyway. Probing user-agent should be for the convience, but selector could help people visiting from *Platform A* for *Platform B* details.

**Comment by afmpjr** [comment]
> @pfeerick No, I only get this image from my Android phone, the screenshot was taken with desktop mode activated
> 
> ![temp_Screenshot_20240305-223821](https://private-user-images.githubusercontent.com/50301201/310353191-8a612acc-818e-4932-aa99-d6f8b1b8d6b7.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk2OTkxMTgsIm5iZiI6MTcwOTY5ODgxOCwicGF0aCI6Ii81MDMwMTIwMS8zMTAzNTMxOTEtOGE2MTJhY2MtODE4ZS00OTMyLWFhOTktZDZmOGIxYjhkNmI3LmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzA2VDA0MjAxOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTcxYWE0ZWFmMWQ1OTQ4NmQ4MDY0ZDcxZTAxNTgxNTQwZTQxNDYyMjg4OGE5NWU3NDQyZmQzYmY2ODE3N2NlNmYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.KESyjUBENcsf-RJUEUwvdDb-THjRUS1mJ0d66hKKd88)
> 
> @k3davis I see you also have the same problem as mine. Have you been able to solve it?
> 
> @alexlamgit How did you do it?

When you access [this page](https://www.warp.dev/?windows_waitlist=true) you should be able to see the e-mail field at the bottom, like this:
![image](https://github.com/warpdotdev/Warp/assets/4296169/26c9f72f-bad1-4a2d-a008-59394b79d688)

**Comment by Dhyfer1** [comment]
> You probably need to access that page from a Windows machine. Or maybe changing the browser agent to a Windows one. 

I think you are right. I'll try changing the user agent, I hadn't thought of this before.

Edit: I'm using kiwi browser, a browser that supports Chrome extensions. I have installed some extensions to change the user agent but so far no extension has been able to show me the window to join the waiting list. I will keep trying

**Comment by kchawla-pi** [comment]
I was having the same issue as you on my phone but I signed in to the waitlist soon as I used my Windows laptop.

**Comment by wneel** [comment]
We are a lot to be subscribed to this,

It is possible to lock the thread until there is a real update ?

**Comment by RiteshChepuri** [comment]
Yeah, can we have a lock on issue until we get a stable way to get the download.

**Comment by Dhyfer1** [comment]
> I was having the same issue as you on my phone but I signed in to the waitlist soon as I used my Windows laptop.

You're right, I couldn't do it from my phone. Like you, I was able to join from my PC. I hope the Windows version arrives soon.

**Comment by dev-saw99** [comment]
I tried using warp inside wsl2, all the functionalities seem to be working fine but seem to lag a bit. Anyone tried using it with WSL2 please share your experience.

**Comment by dannyneira** [comment]
Hey folks, we unlocked the thread so that 👍 's could continue to be recorded. For updates on native Warp for Windows, please [submit your email to get notified here](https://www.warp.dev/windows-terminal)!

Regarding WSL support, we'll be tracking support for Warp for Linux on WSL separately here (that thread includes a lot of discussion on workarounds) https://github.com/warpdotdev/Warp/issues/4240

In the meantime, please avoid spam and WSL troubleshooting comments in the Warp for Windows thread so that users can stay subscribed for updates. We hope to have more news later this year! 🤞

**Comment by Samir-Mulla** [comment]
When warp for windows will release. I am very excited to use it in windows but can I use it in my vscode as a vscode extension while you launch warp.

**Comment by Leiloukou** [comment]
> When warp for windows will release. I am very excited to use it in windows but can I use it in my vscode as a vscode extension while you launch warp. 

I'm confused. Why is this comment so disliked again?

**Comment by pri-me** [comment]
Also the response to it got disliked 😅 although these are valid questions. I don't know about VSCode but IntelliJ just uses the platform terminal. If that is the case for VSCode, too, an extension wouldn't make sense.

**Comment by jakoss** [comment]
Because it's spamming to all observing users. And warp did announce that they are working on this and you can just get notified by email. No need to ask "when it will be ready", that doesn't make anything faster

**Comment by alrz** [comment]
![image](https://github.com/warpdotdev/Warp/assets/3105979/917c265c-2654-41a5-8817-24af57e5e042)

**Comment by Visual917** [comment]
shut up please

**Comment by SleepingRian** [comment]
Please stop talking here, all it does is spam users. Please just fucking wait.

**Comment by enndubyu** [comment]
Is there a way to mute notifications for only the inane comments?

**Comment by memarino92** [comment]
Time to lock this thread again 🔒

**Comment by pri-me** [comment]
is @z-the-kidd an amoc AI or what's wrong with this !d!0t 😅 lock it down mods.

**Comment by Lohenrex** [comment]
> > is @z-the-kidd an amoc AI or what's wrong with this !d!0t 😅 lock it down mods.
> 
> Shut the fuck up nigger. Cotton picking wasn't enough for you to shut the fuck up you negroid?

Guess It takes one to know one. Your username fits you well, you behave like a kid.

**Comment by Dhyfer1** [comment]
> > is @z-the-kidd an amoc AI or what's wrong with this !d!0t 😅 lock it down mods.
> 
> Shut the fuck up nigger. Cotton picking wasn't enough for you to shut the fuck up you negroid?

Hey yo, calm down homie 🫳

**Comment by xorinzor** [comment]
Can we please stop giving this attention?

If everyone just stops replying unless they have something relative to the main issue to add we can stop all the spam.

**Comment by EasternBloxxer** [comment]
> > is @z-the-kidd an amoc AI or what's wrong with this !d!0t 😅 lock it down mods.
> 
> Shut the fuck up nigger. Cotton picking wasn't enough for you to shut the fuck up you negroid?

hello, i am Grincher. how about you take your pompous self-righteousness and shove it where the sun don't shine?

Sincerely,
Grincher.

**Comment by noahzweben** [comment]
Hi all! We’ve begun to scope out our Warp on Windows build. We’re looking for some community feedback on what shells to support. **Please check out this form (only 5 questions!)** [here](https://docs.google.com/forms/d/1ltph1Xhc6hkzIB6USpnIewsBTYUL5EOveSvyBtxnSFk/edit)

**Comment by noahzweben** [comment]
Hi all! We've made progress on our Warp on Windows build and have a **5 question follow-up survey about Windows and Powershell functionality to inform our roadmap** [here](https://forms.gle/xEN6B5XPnzcKYfLy7)!

Thank you for your support!

**Comment by abhishekp106** [comment]
Hey folks! We wanted to share a new blog post that highlights what we've learned building Warp on Windows. If you're interested, take a read here: https://www.warp.dev/blog/building-warp-on-windows.

We're excited to get it out to you all soon!

**Comment by abhishekp106** [comment]
**Warp on Windows is officially live for the general public!** 🎉 

Thank you to everyone who subscribed, shared feedback, and supported us throughout this journey. Your patience and dedication have been incredible—we know this has been a long time coming, and we truly appreciate you sticking with us.

**Download Warp for Windows & learn more**: [warp.dev/windows-terminal](http://warp.dev/windows-terminal)

We’d love for you to spread the word—share your experience on [X](https://x.com/warpdotdev/status/1894749298560045526), [Bluesky](https://bsky.app/profile/warp.dev/post/3lj3jzee5y22x), [LinkedIn](https://www.linkedin.com/posts/warpdotdev_introducing-warp-on-windows-with-this-release-activity-7300515149201756160-s9aQ?utm_source=share&utm_medium=member_desktop&rcm=ACoAABoGMMEBY3g-8K5BDD9b89QHLK6lBe09WlY), or anywhere else!

We're still working on improvements to make the Warp on Windows experience even better. If you hit any bugs, please file a new GitHub issue so we can track.
